ksqldb-functional-tests/src/main/java/io/confluent/ksql/test/driver/AssertExecutor.java

```diff
@@ -122,12 +129,57 @@ private static void assertContent(
     }
   }
 
-  public static void assertStream(final AssertStream assertStatement) {
-    throw new UnsupportedOperationException();
+  public static void assertStream(
+      final KsqlExecutionContext engine,
+      final AssertStream assertStatement
+  ) {
+    assertSourceMatch(engine, assertStatement.getStatement());
+  }
+
+  public static void assertTable(
+      final KsqlExecutionContext engine,
+      final AssertTable assertStatement
+  ) {
+    assertSourceMatch(engine, assertStatement.getStatement());
   }
 
-  public static void assertTable(final AssertTable assertStatement) {
-    throw new UnsupportedOperationException();
+  private static void assertSourceMatch(
+      final KsqlExecutionContext engine,
+      final CreateSource statement
+  ) {
+    final SourceName source = statement.getName();
+    final DataSource dataSource = engine.getMetaStore().getSource(source);
+    final LogicalSchema actual = dataSource.getSchema();
+    final LogicalSchema expected = statement.getElements().toLogicalSchema();
+    if (!actual.equals(expected)) {
+      throw new KsqlException(
+          String.format(
+              "Expected schema does not match actual.%n\tExpected: %s%n\tActual: %s",
+              expected,
+              actual
+          )
+      );
+    }
+
+    final DataSourceType type = dataSource.getDataSourceType();
+    if (statement instanceof CreateStream && type != DataSourceType.KSTREAM) {
+      throw new KsqlException("Expected source " + source + " to be a STREAM but was " + type);
+    } else if (statement instanceof CreateTable  && type != DataSourceType.KTABLE) {
+      throw new KsqlException("Expected source " + source + " to be a TABLE but was " + type);
+    }
+
+    final KsqlTopic topic = dataSource.getKsqlTopic();
+    if (!topic.getKafkaTopicName().equals(statement.getProperties().getKafkaTopic())) {
+      throw new KsqlException("Expected source " + source + " to have kafka topic "
+          + topic.getKafkaTopicName() + " but was " + statement.getProperties().getKafkaTopic());
+    }
+
+    if (!topic.getValueFormat().getFormatInfo().equals(statement.getProperties().getFormatInfo())) {
+      throw new KsqlException("Expected source " + source + " to have value format of "
+          + topic.getValueFormat().getFormatInfo() + " but got "
+          + statement.getProperties().getFormatInfo()
+      );
+    }
```

Comment:
What about other properties that could be specified? For example, timestamp, timestamp format, or wrap single values? If we aren't planning to support validating them in the tester it'd be nice to (1) document this (apologies if I've simply missed it) and (2) fail if they're provided so the user knows they aren't valid rather than thinking they're being checked when they aren't.

Reply:
I gave this some thought during implementation but now that I'm revisiting it, I'm not sure why I decided against having these checks. I'll add them back in